clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -34,19 +39,46 @@ abstract class IndexService[CI <: ClioIndex](
 
   def upsertMetadata(
     indexKey: clioIndex.KeyType,
-    metadata: clioIndex.MetadataType
-  ): Future[UpsertId] = {
-    val updatedMetadata = metadata.withDocumentStatus(
-      metadata.documentStatus.orElse(Some(DocumentStatus.Normal))
-    )
-
-    persistenceService
-      .upsertMetadata(
-        indexKey,
-        updatedMetadata,
-        documentConverter,
-        elasticsearchIndex
-      )
+    metadata: clioIndex.MetadataType,
+    force: Boolean = false
+  )(implicit materializer: Materializer): Future[UpsertId] = {
+    // query by key to see if the document already exists
+    queryMetadataForKey(indexKey)
+      .runFold[Either[Exception, Json]](
+        // If the query doesn't return a document and the document status is not set default to Normal.
+        Right(
+          metadata.documentStatus
+            .fold(metadata.withDocumentStatus(Some(Normal)))(_ => metadata)
+            .asJson
+        )
+      ) { (queryResult, _) =>
+        // if we are not forcing the we want to run validation
+        if (!force) {
+          queryResult match {
+            case Right(existingMetadata) =>
+              // validation will check to make sure no fields are being overwritten
+              validateUpsert(existingMetadata, metadata.asJson)
+            case _ => queryResult
+          }
+        } else {
+          queryResult
+        }
+      }
+      .flatMap[UpsertId] {
+        case Right(updateJson) =>
+          updateJson.as[clioIndex.MetadataType].map { updateMetadata =>
+            persistenceService.upsertMetadata(
+              indexKey,
+              updateMetadata,
+              documentConverter,
+              elasticsearchIndex
+            )
+          } match {
+            case Right(upsertId) => upsertId
+            case Left(failure)   => Future.failed(failure)
+          }
+        case Left(rejection) => Future.failed(rejection)
```

Comment:
We should be able to `flatMap` over the `Either`s to avoid the nested matches / duplicate handling of `Left`